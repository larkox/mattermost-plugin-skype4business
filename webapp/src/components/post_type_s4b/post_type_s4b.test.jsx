// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import PostTypeS4b from './post_type_s4b';

describe('components/post_type_s4b', () => {
    const baseProps = {
        creatorName: 'test creator',
        theme: {id: 'theme_id'},
        post: {
            props: {
                meeting_link: 'https://test.com',
            },
        },
    };

    it('should match snapshot without a meeting topic', () => {
        const wrapper = shallow(<PostTypeS4b {...baseProps}/>);

        baseChecks(wrapper);
        expect(wrapper.find('h1').text()).toEqual('Skype for Business Meeting');
    });

    it('should match snapshot with a meeting topic', () => {
        const props = {...baseProps};
        props.post.props.meeting_topic = 'test topic';
        const wrapper = shallow(<PostTypeS4b {...props}/>);

        baseChecks(wrapper);
        expect(wrapper.find('h1').text()).toEqual('test topic');
    });

    function baseChecks(wrapper) {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('div')).toHaveLength(6);
        expect(wrapper.find('div').first().prop('children')[0]).toEqual('test creator has started a meeting');
        expect(wrapper.find('div').last().text()).toEqual('JOIN MEETING');
        expect(wrapper.find('a').prop('href')).toEqual('https://test.com');
    }
});
