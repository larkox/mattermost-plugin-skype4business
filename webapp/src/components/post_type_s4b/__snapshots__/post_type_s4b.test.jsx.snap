// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/post_type_s4b should match snapshot with a meeting topic 1`] = `
<div>
  test creator has started a meeting
  <div
    style={
      Object {
        "marginLeft": "-20px",
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "borderColor": "#BDBDBF",
          "borderRadius": "4px",
          "borderStyle": "solid",
          "borderWidth": "1px",
          "margin": "5px 0 5px 20px",
          "padding": "2px 5px",
        }
      }
    >
      <div
        style={
          Object {
            "borderLeftColor": "#89AECB",
            "borderLeftStyle": "solid",
            "borderLeftWidth": "4px",
            "padding": "10px",
          }
        }
      >
        <h1
          style={
            Object {
              "fontSize": "16px",
              "fontWeight": "600",
              "height": "22px",
              "lineHeight": "18px",
              "margin": "5px 0 1px 0",
              "padding": "0",
            }
          }
        >
          test topic
        </h1>
        <div>
          <div
            style={
              Object {
                "overflowX": "auto",
                "overflowY": "hidden",
                "paddingRight": "5px",
                "width": "100%",
              }
            }
          >
            <a
              className="btn btn-lg btn-primary"
              href="https://test.com"
              rel="noopener noreferrer"
              style={
                Object {
                  "borderRadius": "4px",
                  "color": undefined,
                  "fontFamily": "Open Sans",
                  "fontSize": "12px",
                  "fontWeight": "bold",
                  "letterSpacing": "1px",
                  "lineHeight": "19px",
                  "marginTop": "12px",
                }
              }
              target="_blank"
            >
              JOIN MEETING
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/post_type_s4b should match snapshot without a meeting topic 1`] = `
<div>
  test creator has started a meeting
  <div
    style={
      Object {
        "marginLeft": "-20px",
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "borderColor": "#BDBDBF",
          "borderRadius": "4px",
          "borderStyle": "solid",
          "borderWidth": "1px",
          "margin": "5px 0 5px 20px",
          "padding": "2px 5px",
        }
      }
    >
      <div
        style={
          Object {
            "borderLeftColor": "#89AECB",
            "borderLeftStyle": "solid",
            "borderLeftWidth": "4px",
            "padding": "10px",
          }
        }
      >
        <h1
          style={
            Object {
              "fontSize": "16px",
              "fontWeight": "600",
              "height": "22px",
              "lineHeight": "18px",
              "margin": "5px 0 1px 0",
              "padding": "0",
            }
          }
        >
          Skype for Business Meeting
        </h1>
        <div>
          <div
            style={
              Object {
                "overflowX": "auto",
                "overflowY": "hidden",
                "paddingRight": "5px",
                "width": "100%",
              }
            }
          >
            <a
              className="btn btn-lg btn-primary"
              href="https://test.com"
              rel="noopener noreferrer"
              style={
                Object {
                  "borderRadius": "4px",
                  "color": undefined,
                  "fontFamily": "Open Sans",
                  "fontSize": "12px",
                  "fontWeight": "bold",
                  "letterSpacing": "1px",
                  "lineHeight": "19px",
                  "marginTop": "12px",
                }
              }
              target="_blank"
            >
              JOIN MEETING
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
